content/blog/greenlake-platform-infrastructure-monitoring-with-apache-skywalking.md

```diff
@@ -0,0 +1,144 @@
+
+cd node_exporter-1.4.0-rc.0.linux-amd64
+
+cp node_exporter /usr/sbin
+
+cat<<EOF >nodeexporter.service
+
+[Unit]
+Description=Service to start the node_exporter on the node
+[Service]
+Type=Simple
+ExecStart=/usr/sbin/node_exporter
+StandardOutput=syslog
+StandardError=syslog
+SyslogIdentifier=%n
+
+[Install]
+WantedBy=multi-user.target
+
+EOF
+
+
+mv nodeexporter.service /etc/systemd/system
+systemctl daemon-reload
+systemctl enable nodexporter.service
+systemctl start nodexporter.service
+```
+
+ 
+
+*  Check the metrics endpoint shows metrics exported from the infrastructure node
+
+```
+curl <http://localhost:9100/metrics>
+```
+
+ 
+
+![](/img/node_metrics.png "Node Metrics")
+
+2. Setup OTel (OpenTelemetry) collector
+
+OTel provides a set of standardized vendor-agnostic SDKs, APIs, and tools for ingesting, transforming, and sending data to an Observability back-end. We can install OTel collector in the same host which exposes metrics or we can create central VM/Bare metal instance to receive telemetry information from several infrastructure instances.
+
+* Install OTel collector
+
+```
+sudo wget <https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v0.54.0/otelcol_0.54.0_linux_amd64.deb>
+
+sudo dpkg -i otelcol_0.54.0_linux_amd64.deb
+
+systemctl status otelcol
+```
+
+ 
+
+**Note:** By default, the otelcol systemd service will be started with the --config=/etc/otelcol/config.yaml option after installation.
+
+* Create a new otel configuration file (say /etc/otelcol/otel-collector-config.yaml) to link the metrics from host nodes to SkyWalking oap. The orange box in below images must be modified to point to correct infrastructure hosts and SkyWalking DNS or IP address.
+
+  
+
+![](/img/otel_collector_configuration.png "OTeL Configuration")
+
+* Change OTel configuration to newly created OTel configuration file
+
+```
+cd /etc/otelcol
+
+sudo nano otelcol.conf
+
+#change the path to new configuration file we created in the above step and save
+
+
+sudo systemctl restart otelcol
+
+sudo journalctl -u otelcol
+```
+
+
+
+3. Validate the infrastructure host metrics are seen in the SkyWalking UI 
+
+![](/img/skywalking_vm_monitoring.png "SkyWalking UI with Infrastructure details")
+
+**Conclusion**
+
+SkyWalking in a popular APM tool used for monitoring and observability of customer workloads. This blog focused on how we could do, infrastructure host monitoring using OpenTelemetry and SkyWalking. There are other blogs which gives insight into how could use it for application level monitoring for distributed microservices based workload. Infrastructure and application level monitoring gives holistic picture of service reliability.
+
+ 
+
+**Reference**
+
+\[Apache SkyWalking](https://skywalking.apache.org/docs/main/v9.3.0/en/concepts-and-designs/overview/)
+
+\[Monitoring and observability](<https://cloud.google.com/architecture/devops/devops-measurement-monitoring-and-observability>
```

Comment:
remove the backslash and remove the "<" and ">" in the URL and make sure to terminate the URL with a parenthesis.